### ISSUE TITLE
Add ability to index and filter by DocID

### ISSUE THREAD

**Issue by gkorland** [RediSearch/RediSearch#1091]

**Comment by mnunberg** [comment]
I don't know what index by docid means here; but we can filter using `INKEYS`

**Comment by MeirShpilraien** [comment]
You can have a tag fields with the docid, why it need to be force or allow it as a specific option?

**Comment by mnunberg** [comment]
That would be a terrible abuse of the tag field, not intuitive, and a memory hog. You're creating a new inverted index for each id.

**Comment by MeirShpilraien** [comment]
You are right, we should have another field type which is a single value tag. I can see how it useful not only to index docid but to index ids in general...

**Comment by mnunberg** [comment]
The problem is that we are an inverted index; value-based indexing would totally mess with how we do things. I suppose people could use numeric indexes (which already have the same issue

In any event, I think we can allow to *filter* (but not *index*) by docid, using the following proposed syntax

```
FILTER $key = "deadbeef"
```

It's trivial to add `FILTER` (or really any "aggregation-only" keyword) into FT.SEARCH

**Comment by jphaugla** [comment]
This is from a customer of mine.

I'm working on my spring boot starter. A complaint from my developers has been that client-side paging hits the hard 1M limit. But server-side paging has no option to pull only keys. So any (client) operation that wants to operate off keys alone (and still be certain we're working outside the 1M limit) pays the cost of pulling the gzip-compressed serialized document from Redis, sending it to the client, decompressing, deserializing, and then discards 99% of the payload. <barf>
I'm digging into the details and not seeing how I can entice FT.AGGREGATE to return the keys. It appears you have to use the LOAD param but in such a scenario you can only pull individual fields… which I think means fields explicitly added to the index. However, the key isn't one of those fields. I'm hoping there's some hook to pull the key.
The goal is to return the keys and save the pain described above.

**Comment by mnunberg** [comment]
Is it possible for them to use FT.SEARCH? are they doing aggregation solely for the cursor feature? 1.6 allows cursors to be used with search as well.

**Comment by rnbWarden** [comment]
Yes, @mnunberg we're using aggregation solely for the cursor feature. We're still on 1.4 at the present time. If 1.6 provides cursors for FT.SEARCH then this should sufficiently handle our issue.
